Yoast/yoast-components
# Implement collapsible panel component for classic editor sidebar

## Summary

This PR can be summarized in the following changelog entry:

* Adds a collapsible panel component: Collapsible and CollapsibleStateless.
* Refactored AnalysisCollapsible and AnalysisCollapsibleStateless to use the new collapsible.
* Adds a new button component: IconsButton. It supports a prefix and suffix SvgIcon besides text.
* Like the old collapsible you can vary the header level (h1, h2, h3, etc). But you always have a header now.
* Uses the IconsButton inside the header, but with different icons when open or closed.
* We do not have a WordPress sidebar widget. So this is not actually used in one.... yet.

## Relevant technical choices:

* The stateful variant manages the `isOpen` and `Header`. The first is through the click of the button. The second is with the variable `headingLevel`.
* Another stateful difference is the children are wrapped in a styled `div` to give them a padding. Basically the AnalysisCollapsible does not want this and uses a `ul` container.

## Test instructions

This PR can be tested by following these steps:

* `yarn install`, `yarn start`, then go to `http://localhost:3333/` and check out the new tab: `Sidebar Collapsible`.
* Check the `Content analysis` tab too. Does this look and behave the same as before?
* Link to `wordpress-seo` and check the above in there!

Fixes #436
